app/components/submission/ManuscriptMetadata/SubjectAreaDropdown.test.js

```diff
@@ -0,0 +1,120 @@
+import React from 'react'
+import { shallow, mount } from 'enzyme'
+import theme from '@pubsweet/elife-theme'
+
+import SubjectAreaDropdown from './SubjectAreaDropdown'
+
+const onChange = jest.fn()
+
+const MyDropdown = (
+  <SubjectAreaDropdown
+    label="My label"
+    name="My name"
+    onBlur={jest.fn()}
+    onChange={onChange}
+    savedValues={[]}
+    theme={theme}
+  />
+)
+
+describe('SubjectAreaDropdown component', () => {
+  it('renders label & select', () => {
+    const wrapper = shallow(MyDropdown).dive() // dive is necessary because SubjectAreaDropdown is wrapped by withTheme
+    expect(wrapper.contains('My label')).toBe(true)
+    expect(wrapper.find('[inputId="subject-area-select"]').exists()).toEqual(
+      true,
+    )
+    expect(wrapper.contains('No more than two subject areas')).toBe(false)
+  })
+
+  describe('Integration with react-select', () => {
+    let wrapper, selectWrapper, selectInput
+
+    beforeEach(() => {
+      wrapper = mount(MyDropdown)
+      selectWrapper = wrapper.find('Select')
+      selectInput = selectWrapper.find('input')
+    })
+
+    const typeSubjectArea = letters => {
+      // We need 2 steps to programmatically simulate typing in a letter
+      // 1. Add a letter to the value of the input field (does not cause the dropdown to appear by itself)
+      selectWrapper.props().onInputChange(letters)
+      // 2. Open the dropdown
+      selectInput.simulate('change')
+    }
+
+    it('focusing causes the dropdown to open & display every option', () => {
+      typeSubjectArea('') // typing nothing into the field seems to be the only way to programmatically focus react-select)
+      expect(wrapper.find('Menu').exists()).toEqual(true)
+      expect(wrapper.find('Option')).toHaveLength(17)
```

Comment:
Hm, since you have `eLifeOptions` as a global variable in `SubjectAreaDropdown.js`, maybe you can export it from there and not hardcode 17 here?

Reply:
I figured exporting `eLifeOptions` would only be a temporary solution, if we're going to call the eLife API at some point.

But yeah, either way, I probably shouldn't hard code 17 into here.

Do you have a suggestion for how I could test that all of the options are rendered, rather than just a subset (e.g. when it's filtered by a given letter), even when the list is dynamically generated?

Reply:
Not really, if this is going to change I assume it's fine for now (updating the component usually results in updating the tests as well). I would probably try to query the eLife API here once we do this in the component itself.